app/src/test/java/com/sebastianvm/mango/ui/example/ExampleViewModelTest.kt

```diff
@@ -0,0 +1,43 @@
+package com.sebastianvm.mango.ui.example
+
+import com.google.common.truth.Truth.assertThat
+import com.sebastianvm.mango.BaseTest
+import com.sebastianvm.mango.FakeProvider
+import com.sebastianvm.mango.data.FakeJobRepository
+import com.sebastianvm.mango.database.models.JobEntity
+import kotlinx.coroutines.test.runTest
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.MethodSource
+
+class ExampleViewModelTest : BaseTest() {
+
+    private lateinit var jobRepository: FakeJobRepository
+
+    @BeforeEach
+    fun beforeEach() {
+        jobRepository = FakeProvider.jobRepository
+    }
+
+    private fun generateViewModel(): ExampleViewModel {
+        return viewModelForTests {
+            ExampleViewModel(
+                initialState = ExampleState(),
+                jobRepository = jobRepository
+            )
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("com.sebastianvm.mango.FakeProvider#jobEntityListProvider")
+    fun `init sets state correctly and subscribes to changes`(jobEntityList: List<JobEntity>) =
```

Comment:
I'm guessing this is a JUnit5 thing?

Reply:
Yeah, main reason I wanted to use JUnit5, we can run tests with different parameters, which makes it easier to test different cases. We can probably use our FakeProvider for many cases so out tests will have little need to initialize objects